spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java

```diff
@@ -62,23 +62,27 @@ public ModuleLauncher(ModuleResolver moduleResolver) {
 	 * The format of each module must conform to the <a href="http://www.eclipse.org/aether">Aether</a> convention:
 	 * <code>&lt;groupId&gt;:&lt;artifactId&gt;[:&lt;extension&gt;[:&lt;classifier&gt;]]:&lt;version&gt;</code>
 	 *
-	 * To pass arguments to a module, prefix with the module name and a dot. The arg name will be de-qualified and passed along.
-	 * For example: <code>--org.springframework.cloud.stream.module:time-source:1.0.0.BUILD-SNAPSHOT.bar=123</code> becomes <code>--bar=123</code> and is only passed to the 'org.springframework.cloud.stream.module:time-source:1.0.0.BUILD-SNAPSHOT' module.
-	 *
-	 * @param modules a list of modules
-	 * @param args a list of arguments, prefixed with the module name
+	 * @param moduleLaunchRequests a list of modules with their (unqualified) arguments
+	 */
+	public void launch(Iterable<ModuleLaunchRequest> moduleLaunchRequests) {
```

Comment:
We need to decide on the contract with regards to ordering of the modules. Deployment order should be starting with the sink and ending with the source. However, I don't think we are sending in that order, at least not consistently. The other question is if it's more intuitive to make the contract so that this expects the natural order instead (source in the first position, sink in the last). That is more intuitive for the user. It would mean, however, that we need to deploy what is passed in reverse order, and thus `Iterable` might not be the most convenient type for this method parameter.